.github/workflows/codeql-analysis.yml

```diff
@@ -14,11 +14,17 @@ on:
   schedule:
     - cron: '0 13 * * 6'
 
+permissions: {}
```

Comment:
Explicit `none` permissions is more restrictive than the repo-level and org-level settings provided by the GitHub GUI. This explicitness will also mitigate the impact of misconfiguration from that aforementioned GUI.

Will still adjust via the GitHub GUI as well as part of a defence-in-depth strategy.